File: 7.整数反转/7.整数反转.py
#
# @lc app=leetcode.cn id=7 lang=python3
#
# [7] 整数反转
#

# @lc code=start
class Solution:
    def reverse(self, x: int) -> int:
        if -2**31 < x < 2**31-1:
            tmp = str(x).replace('-', '')
            tmp = int(tmp[::-1])
            if tmp > 2**31-1:
                return 0

            if x > 0:
                return tmp
            else:
                return -tmp

        else:
            return 0
# @lc code=end

File: 7.整数反转/7.整数反转.cs
/*
 * @lc app=leetcode.cn id=7 lang=csharp
 *
 * [7] 整数反转
 */

// @lc code=start
public class Solution
{
    public int Reverse(int x)
    {
        long tmp = 0;

        int inputX = x;

        while (inputX != 0)
        {
            tmp = tmp * 10 + inputX % 10;
            inputX = inputX / 10;
        }

        if (tmp < int.MinValue || tmp > int.MaxValue)
        {
            return 0;
        }

        return (int)tmp;
    }
}
// @lc code=end

